SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h

```diff
@@ -127,6 +127,25 @@ class SPATIALGDK_API USpatialNetDriver : public UIpNetDriver
 	void StartIgnoringAuthoritativeDestruction() { bAuthoritativeDestruction = false; }
 	void StopIgnoringAuthoritativeDestruction() { bAuthoritativeDestruction = true; }
 
+#if !UE_BUILD_SHIPPING
+	uint8 GetNextReliableRPCId(AActor* Actor, ESchemaComponentType RPCType, UObject* TargetObject);
+	void OnReceivedReliableRPC(AActor* Actor, ESchemaComponentType RPCType, FString WorkerId, uint8 RPCId, UObject* TargetObject, UFunction* Function);
+
+	struct FReliableRPCId
+	{
+		FReliableRPCId() = default;
+		FReliableRPCId(FString InWorkerId, uint32 InRPCId, FString InRPCTarget, FString InRPCName) : WorkerId(InWorkerId), RPCId(InRPCId), LastRPCTarget(InRPCTarget), LastRPCName(InRPCName) {}
+
+		FString WorkerId;
+		uint32 RPCId = 0;
+		FString LastRPCTarget;
+		FString LastRPCName;
+	};
+	using FRPCTypeToReliableRPCIdMap = TMap<ESchemaComponentType, FReliableRPCId>;
+	// Per actor, maps from RPC type to the reliable RPC index used to detect if reliable RPCs go out of order.
+	TMap<TWeakObjectPtr<AActor>, FRPCTypeToReliableRPCIdMap> ReliableRPCIdMap;
```

Comment:
Currently, this will leak. We could swipe it for deleted actors on tick, or once in a while, but I figured that may be a premature optimization for a debug functionality.